atc/creds/vault/vault_test.go

```diff
@@ -147,5 +147,30 @@ var _ = Describe("Vault", func() {
 				Expect(err).To(BeNil())
 			})
 		})
+
+		Context("allowRootPath", func() {
+			JustBeforeEach(func() {
+				v = &vault.Vault{
+					SecretReader: msr,
+					Prefix:       "/concourse",
+				}
+
+				variables = creds.NewVariables(v, "team", "pipeline", true)
+			})
+
+			It("should not get secret from root", func() {
```

Comment:
should probably have two examples: one showing it can and one showing it can't. this one seems to say it can't but then demonstrates that it does (because it's given `true` in the `JustBeforeEach`)